tchannel/tornado/peer.py

```diff
@@ -578,7 +629,9 @@ def get(self, hostport):
         """
         assert hostport, "hostport is required"
         if hostport not in self._peers:
-            self._peers[hostport] = self.peer_class(self.tchannel, hostport)
+            self.add(hostport)
+            return self._peers[hostport]
+
         return self._peers[hostport]
```

Comment:
drop the return from the if statement. you're already returning the same result on the next line.